Parser/Node.cs

```diff
@@ -44,12 +55,43 @@ public record NWriteStmt (bool NewLine, NExpr[] Exprs) : NStmt {
    public override T Accept<T> (Visitor<T> visitor) => visitor.Visit (this);
 }
 
+// A Read statement
+public record NReadStmt (Token Name, Token[] Names) : NStmt {

```

Comment:
Why do you need **Name**? That will always be the **read** token and it can be dropped, you don't need to store it in the NReadStmt record.